types_test.go

```diff
@@ -49,3 +50,123 @@ func TestFeatureMap(t *testing.T) {
 		}
 	})
 }
+
+func TestMetadata(t *testing.T) {
+	t.Run("different Metadata are not equal", func(t *testing.T) {
+		a := Metadata{
+			"abra":       1,
+			"bulbasaur":  "TWO",
+			"charmander": false,
+		}
+		b := Metadata{
+			"abra":       2,
+			"bulbasaur":  "one",
+			"charmander": true,
+		}
+		c := Metadata{
+			"abra":      1,
+			"bulbasaur": "TWO",
+		}
+
+		if a.Equals(b) {
+			t.Error("A and B are not equal, but are.")
```

Comment:
But are?

Reply:
It's expected that they are not equal, but the error is that they are equal :D

Reply:
This looks to me like a double negation :P

Reply:
It is! :) cause its a negative test case

Reply:
okay okay, I'll reword this :D